### PR TITLE
feat: node version lock

### PR DESCRIPTION
**Pull request by dennisvankekem**
changes:
- enforce specific node version per project  (v16)
- added some default formatters
- added .nvmrc

issue:
- was not able to bump node version in tools Dockerfile to node v18, therefore the project is still using v16. Might troubleshoot with the team.

Notice: 
Merging this branch will enforce the current node version (v16). That means that without the right version you will not be able to run or build the project locally. To overcome this issue you should install NVM  with the following command: 

`curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash`

Within the scope of the project you can use the command `nvm use` to automatically switch to the required node version as specified in the `.nvmrc` under the condition that said node version is installed. This process is quite cumbersome but can be automated be adding the script below to your `.zshrc` file. This will automatically switch to the right version of node when you open a project and will also install it if not present on your machine.

```
# place this after nvm initialization!
autoload -U add-zsh-hook
load-nvmrc() {
  local node_version="$(nvm version)"
  local nvmrc_path="$(nvm_find_nvmrc)"

  if [ -n "$nvmrc_path" ]; then
    local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")

    if [ "$nvmrc_node_version" = "N/A" ]; then
      nvm install
    elif [ "$nvmrc_node_version" != "$node_version" ]; then
      nvm use
    fi
  elif [ "$node_version" != "$(nvm version default)" ]; then
    echo "Reverting to nvm default version"
    nvm use default
  fi
}
add-zsh-hook chpwd load-nvmrc
load-nvmrc
``` 

To change to global version of node use the command `nvm alias default [NODE_VERSION]`. Don't forget to restart your terminal session after applying any of the above.

Also, it is a best practice to let node decide on the npm version, do not install your own version of npm unless you have a good reason to (please consult it with me, I'm curious what it might be).

### CODE REVIEW

**Comment by dennisvankekem** [comment]
Project has been tested on DO test environment

**Review by j-zimnowoda — COMMENTED** [review]
Nice! Just try to implement script in .husky/* as @Morriz suggested

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v18.12.1
```
Comment (j-zimnowoda) [inline .nvmrc:1]: should be v16 right ?

Comment (j-zimnowoda) [reply .nvmrc:1]: v16.16.0 (LTS)

**Comment by j-zimnowoda** [comment]
Hey Dennis,
I noticed that if a given node version is not installed the post-checkout hook tries to install but it fails with the following error message 
```
Sourcing nvm from ~/.nvm
current node version: v18.12.1
husky - post-checkout hook exited with code 3 (error)
```

Any changes to improve this error log or at least to ask a user to install the missing node version ?

File: .nvmrc (hunk 1 above)
Comment (dennisvankekem) [reply .nvmrc:1]: Nice catch! Yes, it should! I'll change it right away

**Comment by dennisvankekem** [comment]
> 

Very strange, this did not happened when I tested it. I'm rather curious what is causing this

**Review by j-zimnowoda — APPROVED** [review]

**Pull request merged**